hudi-client/src/main/java/org/apache/hudi/table/HoodieCopyOnWriteTable.java

```diff
@@ -109,7 +109,7 @@ public HoodieCopyOnWriteTable(HoodieWriteConfig config, JavaSparkContext jsc) {
         Tuple2<String, String> partitionDelFileTuple = iter.next();
         String partitionPath = partitionDelFileTuple._1();
         String delFileName = partitionDelFileTuple._2();
-        Path deletePath = new Path(new Path(basePath, partitionPath), delFileName);
+        Path deletePath = FSUtils.getPartitionPath(FSUtils.getPartitionPath(basePath, partitionPath), delFileName);
```

Comment:
Does this simply need a `new Path(FSUtils.getPartitionPath(basePath, partitionPath), delFileName)` ? We are overloading calling of `FSUtils.getPartitionPath`. for somethings that's not a partitionpath? 

Also can we please ensure there is a JIRA for fix.. or we follow the `[MINOR] ...` convention?

@cdmikechen @yanghua wdyt

Reply:
> Does this simply need a `new Path(FSUtils.getPartitionPath(basePath, partitionPath), delFileName)` ? We are overloading calling of `FSUtils.getPartitionPath`. for somethings that's not a partitionpath?

I saw `FsUtils#getPartitionPath` can only avoid the second arg to be null. However, the outer `new Path(..., delFileName);` The `delFileName` can not be null?

> Also can we please ensure there is a JIRA for fix.. or we follow the `[MINOR] ...` convention?
> 

Agree, I have mentioned in the approvement comment. Not sure whether `[MINOR]`  is a normal prefix if there is not a Jira issue to track. IMO, We should clearly record where. Or does it exist but I don't know?

Reply:
@vinothchandar
If by name, it should be better to just modify`FSUtils.getPartitionPath(basePath, partitionPath)`. Maybe I should revert and put another PR?
In hadoop 2.7+, I found that `new Path()` api use more stringent checks than ever before, if user use null to use `new Path(path, null)` in hadoop2.7-, it doesn't report error. 
If it's for API compatibility, is it better to adjust its name to `buildMultiPath()` or else?So that all similar methods can be used, thus to ensure that no exception will happen.

Reply:
@vinothchandar
If by name, it should be better to just modify`FSUtils.getPartitionPath(basePath, partitionPath)`. Maybe I should revert and put another PR?
In hadoop 2.7+, I found that `new Path()` api use more stringent checks than ever before, if user use null to use `new Path(path, null)` in hadoop2.7-, it doesn't report error. 
If it's for API compatibility, is it better to adjust its name to `buildMultiPath()` or else?So that all similar methods can be used, thus to ensure that no exception will happen.

Reply:
I think we can do a follow up small fix here.. confirm `delFileName` cannot be null and then remove the outer `FSUtils.getPartitionPath`. This one can have `[MINOR]` prefix IMO